isisd/isis_snmp.c

```diff
@@ -2546,7 +2546,8 @@ static uint8_t *isis_snmp_find_isadj(struct variable *v, oid *name,
 		return SNMP_INTEGER(adj->threeway_state);
 
 	case ISIS_ISADJ_NEIGHSNPAADDRESS: {
-		const char *snpa = (char *)snpa_print(adj->snpa);
+		static char snpa[ISO_SYSID_STRLEN];
+		snprintfrr(snpa, ISO_SYSID_STRLEN, "%pIS", adj->snpa);
```

Comment:
I think that the ring buffer was implemented because it relies on a global function i.e. `isis_format_id()` which is used in many places through `snpa_print()`, `sysid_print()` and `rawlspid_print()` functions. Even if this ring buffer was not the perfect way to manage this, it avoids trouble in printing when `isis_format_id()`  function is called by many others function in a more or less parallel or concurrent way. However, with the new pretty print format, these functions are no more needed and each call to printfrr function with the new '%pXX` used its own buffer. Thus, ring buffer is no more necessary.

Now, regarding the SNMP stuff, the function `isis_snmp_find_isadj()` which call the `snprintfrr()` seems strange. Indeed, it print the snpa value in a buffer, compute the length and return the buffer as a table of unint8_t. If you look 10 lines below, the same function return directly the adjacency system ID. Thus, I don't understand why the system ID is return directly as the uint8_t table and the adjacency SNPA as a table of char. I suspect a bug here, but, perhaps I'm wrong. I'm not an SNMP expert.

Regarding your proposal, honestly, I don't know how SNMP is handle in FRR, and more precisely how the  `isis_snmp_find_isadj()` is called. Thus, using a dynamic buffer e.g. with strdup() will lead into a memory leak as I don't know where, in the snmp code, I should free the previously allocated memory.

What I could do, is to use the a similar static ring buffer in the IS-IS SNMP code if you think that a simple static buffer is not sufficient.

Reply:
> What I could do, is to use the a similar static ring buffer in the IS-IS SNMP code if you think that a simple static buffer is not sufficient.

I agree with you. A similar static ring buffer would keep the same behavior. Any problem with the ring buffer should be addressed in another pull request.

Reply:
I added the ring buffer for snmp stuff, but, IMHO, it is not necessary.